### PR TITLE
Bump the versions of Packer, Terraform, and terraform-docs

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request updates the following dependencies:
- Packer from [1.9.2 to 1.9.5](https://github.com/hashicorp/packer/compare/v1.9.2...v1.9.5)
- Terraform from [1.5.5 to 1.5.7](https://github.com/hashicorp/terraform/compare/v1.5.5...v1.5.7)
- terraform-docs from [v0.16.0 to v0.17.0](https://github.com/terraform-docs/terraform-docs/compare/v0.16.0...v0.17.0)

It also adds comments explaining the self-imposed version constraints for Packer and Terraform.

## 💭 Motivation and context ##

It's good to stay up-to-date.

## 🧪 Testing ##

Automated tests pass. I used this branch in a test branch of [cisagov/skeleton-tf-module] and verified that the workflow ran successfully.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

[cisagov/skeleton-tf-module]: https://github.com/cisagov/skeleton-tf-module

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -58,8 +58,10 @@ runs:
     - id: gosec
       run: echo "version=v2.16.0" >> $GITHUB_OUTPUT
       shell: bash
+    # We must remain on 1.9 due to HashiCorp's decision to change the license
+    # of Packer from MPL 2.0 to BSL starting with 1.10.0.
```
Comment (dav3r) [inline action.yml]: Did I miss a legal thang that stated that we are unable to use BSL-licensed products for our use case?

**Review by jsf9k — APPROVED** [review]

File: action.yml (hunk 1 above)
Comment (mcdonnnj) [reply action.yml]: No. I should probably reword this a bit as we're not strictly prohibited from using BSL-licensed products. However, there are the following considerations:
- The Business Source License is not considered an Open Source license [per SPDX](https://spdx.org/licenses/BUSL-1.1.html).
- There are concerns about compatibility with Terraform 1.6+ with non-HashiCorp tooling for Terraform as a result of the license change.
- We might want to switch to [OpenTofu](https://opentofu.org/) in the future and there is only guaranteed compatibility with Terraform <1.6.
-  The [CNCF](https://www.cncf.io/) [recommends considering alternative components](https://github.com/cncf/foundation/blob/main/source-available-recommendations.md#recommendations) if a project uses a component under the BUSL-1.1 (among others).

Based on the above and team conversations (off GitHub) I was under the impression that we had agreed to remain on versions before the license change. If I misunderstood something I'd like to have a team chat about it but I'm also happy to improve the wording to satisfy any concerns about clarity if the version constraint is amenable.

Comment (jsf9k) [reply action.yml]: @mcdonnnj - We should definitely flesh out the code comment with the content of [your comment](https://github.com/cisagov/setup-env-github-action/pull/71#discussion_r1440094473).

Comment (dav3r) [reply action.yml]: I know we discussed this and I think we are all on the same page, but I still think that we should come to an "official" team consensus and plan our path forward at our next team meeting.  I agree with you both about adding more detail to the code comments in this PR. 👍

Comment (mcdonnnj) [reply action.yml]: Please let me know if https://github.com/cisagov/setup-env-github-action/pull/71/commits/afef354747f6e6690f1feeef2558490eaf569005 works for both of you.

Comment (dav3r) [reply action.yml]: That looks solid to me, thanks!

@jsf9k please resolve this conversation if you agree.

**Review by dav3r — APPROVED** [review]
👍 👍 👍

**Review by jsf9k — APPROVED** [review]

**Pull request merged**